tests/firecrackerdrivertest.c

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (C) 2021, Amazon, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library;  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include <config.h>
+
+#include "testutils.h"
+
+#include "virlog.h"
+
+#ifdef WITH_FC
+
+#include "vircommand.h"
+
+# define VIR_FROM_THIS VIR_FROM_NONE
+
+VIR_LOG_INIT("tests.firecrackerdrivertest");
+
+static int
+testDriverConnection(const void *opaque G_GNUC_UNUSED)
+{
+    int ret = 0;
+
+    virConnectPtr conn = virConnectOpen("fc:///system");
+
+    if (conn == NULL)
+        ret = -1;
+
+    if (virConnectClose(conn) < 0)
+        ret = -1;
+
+    return ret;
+}
+
+static void
+downloadKernelAndRootFS(void)
```

Comment:
ideally helpers should be placed in another file. But since it's only one, just move it to the top of the file so that helpers and tests are not interleaved